src/EA.Weee.RequestHandlers/AatfReturn/Obligated/AddObligatedReceivedHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace EA.Weee.RequestHandlers.AatfReturn.Obligated
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using EA.Prsd.Core.Mediator;
+    using EA.Weee.Domain.AatfReturn;
+    using EA.Weee.RequestHandlers.Security;
+    using EA.Weee.Requests.AatfReturn.ObligatedReceived;
+
+    internal class AddObligatedReceivedHandler : IRequestHandler<AddObligatedReceived, bool>
+    {
+        private readonly IWeeeAuthorization authorization;
+        private readonly IAddObligatedReceivedDataAccess obligatedReceivedDataAccess;
+
+        public AddObligatedReceivedHandler(IWeeeAuthorization authorization,
+            IAddObligatedReceivedDataAccess obligatedReceivedDataAccess)
+        {
+            this.authorization = authorization;
+            this.obligatedReceivedDataAccess = obligatedReceivedDataAccess;
+        }
+
+        public async Task<bool> HandleAsync(AddObligatedReceived message)
+        {
+            authorization.EnsureCanAccessExternalArea();
+
+            var aatfWeeReceived = new WeeeReceived(
```

Comment:
Just for info, I added a IGenericDataAccess (GenericDataAccess implementation) so that we don't always have to create specific data access. One of the methods take a ISpecification which can be a Expression.